Subject: [PATCH] Inventory Junking
---
Index: src/engine/net/client/ClientMessagePump.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/engine/net/client/ClientMessagePump.java b/src/engine/net/client/ClientMessagePump.java
--- a/src/engine/net/client/ClientMessagePump.java	(revision 2ff0c9b938a9f96e8d185372c2a4b90716b9dc18)
+++ b/src/engine/net/client/ClientMessagePump.java	(revision fb2465ccfd1196b9cff0f06642f9c5948f8fad5f)
@@ -562,6 +562,9 @@
 
         if (i.isCanDestroy())
             if (itemManager.delete(i) == true) {
+                int value = i.getItemBase().getBaseValue();
+                sourcePlayer.getCharItemManager().addGoldToInventory(value,false);
+                sourcePlayer.getCharItemManager().updateInventory();
                 Dispatch dispatch = Dispatch.borrow(sourcePlayer, msg);
                 DispatchMessage.dispatchMsgDispatch(dispatch, DispatchChannel.SECONDARY);
             }
