Subject: [PATCH] slot NPC fix
---
Index: src/engine/net/client/handlers/ActivateNPCMsgHandler.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/engine/net/client/handlers/ActivateNPCMsgHandler.java b/src/engine/net/client/handlers/ActivateNPCMsgHandler.java
--- a/src/engine/net/client/handlers/ActivateNPCMsgHandler.java	(revision 2e5892ff32e48aab433f8a24a3d1774bd693ccd9)
+++ b/src/engine/net/client/handlers/ActivateNPCMsgHandler.java	(revision 35e18e4186f6d52beeb68cb95a25b187b12ae025)
@@ -144,18 +144,6 @@
                     if (contract.getContractID() == 865)//siege engineer
                         override = true;
                 }
-                if(building.getBlueprint().getBuildingGroup().equals(Enum.BuildingGroup.SHACK)){
-                    if(contract.getContractID() == 830)//merchant
-                        override = true;
-                }
-                if(building.getBlueprint().getBuildingGroup().equals(Enum.BuildingGroup.SHRINE)){
-                    if(contract.getContractID() == 252637)//votary
-                        override = true;
-                }
-                if(building.getBlueprint().getBuildingGroup().equals(Enum.BuildingGroup.BULWARK)){
-                    if(contract.getContractID() == 839)//artillery captain
-                        override = true;
-                }
                 if(override == false) {
                     Logger.error("failed at override with contract: " + contract.getContractID());
                     return false;
Index: src/engine/objects/Blueprint.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/engine/objects/Blueprint.java b/src/engine/objects/Blueprint.java
--- a/src/engine/objects/Blueprint.java	(revision 2e5892ff32e48aab433f8a24a3d1774bd693ccd9)
+++ b/src/engine/objects/Blueprint.java	(revision 35e18e4186f6d52beeb68cb95a25b187b12ae025)
@@ -319,7 +319,7 @@
         if (this.maxSlots <= 1 && this.buildingGroup.equals(BuildingGroup.TOL) == false)
             return maxSlots;
 
-        if (this.maxRank == 1 && currentRank == 1&& this.buildingGroup.equals(BuildingGroup.TOL) == false)
+        if (this.maxRank == 1 && currentRank == 1 && this.buildingGroup.equals(BuildingGroup.TOL) == false)
             return getMaxSlots();
 
         switch (currentRank) {
@@ -335,13 +335,11 @@
                 break;
             case 6:
             case 7:
-                availableSlots = 3;
-                break;
             case 8:
                 availableSlots = 3;
                 break;
             default:
-                availableSlots = 0;
+                availableSlots = 1;
                 break;
         }
         if(this.buildingGroup.equals(BuildingGroup.TOL)){
